### PR TITLE
Added "weaver purge" command.

### PR DESCRIPTION
**Pull request by mwhittaker**
## Example

```console
$ weaver multi purge
WARNING: You are about to delete the following directories used to store logs
and data for local Service Weaver applications. This data will be deleted
immediately and irrevocably. Are you sure you want to proceed?

    - /tmp/serviceweaver/logs/weaver-multi
    - /usr/local/google/home/mwhittaker/.local/share/serviceweaver/multi_registry

Enter (y)es to continue: y

Deleting /tmp/serviceweaver/logs/weaver-multi... ✅
Deleting /usr/local/google/home/mwhittaker/.local/share/serviceweaver/multi_registry... ✅
```

## Overview

This PR introduces `weaver single purge` and `weaver multi purge` commands that delete all logs and data produced by locally run Service Weaver applications. In the future, we may also want to kill lingering processes. For example, we could kill the gke-local controller and distributors.

## Motivation

The motivation for this PR is the same as #220. We're seeing some issues (e.g., #179, #154) that I think may be cause by version issues. Users run `weaver multi logs` on logs produced by a different version of Service Weaver, for example. `weaver purge` makes it easier to debug these kinds of issues, as we can tell people to run `weaver purge` before reproducing their reported bug.

### CODE REVIEW

File: internal/tool/multi/multi.go [hunk 1]
```diff
@@ -53,5 +61,13 @@ var (
 		"status":    status.StatusCommand("weaver multi", defaultRegistry),
 		"metrics":   status.MetricsCommand("weaver multi", defaultRegistry),
 		"profile":   status.ProfileCommand("weaver multi", defaultRegistry),
+		"purge":     tool.PurgeCmd(purgeSpec),
 	}
 )
+
+func must[T any](x T, err error) T {
```
Comment (rgrandl) [inline internal/tool/multi/multi.go:68]: can you add a comment?

**Review by rgrandl — COMMENTED** [review]
The PR looks great, I have some thoughts though. Why do we need a command to delete only the directories? What if someone has things running, and by mistake runs the purge command?

I was wondering if we should have a kill/stop/clear/reset command instead. Or rename purge to one of these names, and later on we can also stop processes.

File: internal/tool/multi/multi.go (hunk 1 above)
Comment (mwhittaker) [reply internal/tool/multi/multi.go:73]: Done.

**Comment by mwhittaker** [comment]
> The PR looks great, I have some thoughts though. Why do we need a command to delete only the directories? What if someone has things running, and by mistake runs the purge command?
> 
> I was wondering if we should have a kill/stop/clear/reset command instead. Or rename purge to one of these names, and later on we can also stop processes.

Thanks Robert! We chatted offline. We'll stick with the name "purge" and make it kill processes as well. I also strengthened the warning message.

File: internal/tool/multi/multi.go [hunk 2]
```diff
@@ -53,5 +61,18 @@ var (
 		"status":    status.StatusCommand("weaver multi", defaultRegistry),
 		"metrics":   status.MetricsCommand("weaver multi", defaultRegistry),
 		"profile":   status.ProfileCommand("weaver multi", defaultRegistry),
+		"purge":     tool.PurgeCmd(purgeSpec),
 	}
 )
+
+// must is a generic implementation of the Go must idiom [1, 2]. It panics if
```
Comment (rgrandl) [inline internal/tool/multi/multi.go:68]: very nice. Thanks for the comment, I learned something new too.

File: internal/tool/single/single.go [hunk 3]
```diff
@@ -36,14 +36,32 @@ var (
...
 	Commands = map[string]*tool.Command{
 		"status":    status.StatusCommand("weaver single", defaultRegistry),
 		"dashboard": status.DashboardCommand(dashboardSpec),
 		"metrics":   status.MetricsCommand("weaver single", defaultRegistry),
 		"profile":   status.ProfileCommand("weaver single", defaultRegistry),
+		"purge":     tool.PurgeCmd(purgeSpec),
 	}
 )
 
+// must is a generic implementation of the Go must idiom [1, 2]. It panics if
```
Comment (rgrandl) [inline internal/tool/single/single.go:53]: can we put this method somewhere in a shared directory? Right now is duplicated among single, multi.

File: runtime/tool/purge.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+)
+
+// PurgeSpec configures the command returned by PurgeCmd.
+type PurgeSpec struct {
```
Comment (rgrandl) [inline runtime/tool/purge.go:28]: Maybe add a TODO here; probably we will also specify some reqular expression or smthg in the `PurgeSpec` based on which we will kill all processes/subprocesses for each deployer?

**Review by rgrandl — APPROVED** [review]
Nice!

File: internal/tool/single/single.go (hunk 3 above)
Comment (mwhittaker) [reply internal/tool/single/single.go]: Done.

File: runtime/tool/purge.go (hunk 4 above)
Comment (mwhittaker) [reply runtime/tool/purge.go:28]: Done. Good idea.

**Review by mwhittaker — COMMENTED** [review]
Thanks Robert!

**Pull request merged**